servicetalk-client-api/src/main/java/io/servicetalk/client/api/ConnectionFactory.java

```diff
@@ -31,7 +34,9 @@
      * Creates and asynchronously returns a connection.
      *
      * @param address to connect.
+     * @param observer {@link TransportObserver} that provides visibility into transport events associated with a new
+     * connection.
      * @return {@link Single} that emits the created connection.
      */
-    Single<C> newConnection(ResolvedAddress address);
+    Single<C> newConnection(ResolvedAddress address, @Nullable TransportObserver observer);
```

Comment:
IIUC, expected usage is to add a `ConnectionFactory` for a client that introduces a `TransportObserver` for the call.
What is the guidance for such factories if there is already an observer passed to them?

Reply:
Good question!
I plan to add a `BiTransportObserver` implementation later that will merge two observers into one, when necessary. Something similar to [BiDnsQueryLifecycleObserverFactory](https://github.com/netty/netty/blob/4.1/resolver-dns/src/main/java/io/netty/resolver/dns/BiDnsQueryLifecycleObserverFactory.java) in netty.

Reply:
Added `TransportObserverConnectionFactoryFilter` that uses `BiTransportObserver` internally, ptal.